examples/platform/esp32/lock/BoltLockManager.cpp

```diff
@@ -431,11 +431,8 @@ bool BoltLockManager::SetUser(chip::EndpointId endpointId, uint16_t userIndex, c
     for (size_t i = 0; i < totalCredentials; ++i)
     {
         mCredentials[userIndex][i] = credentials[i];
-        // TODO: Why are we modifying the passed-in credentials?
-        // https://github.com/project-chip/connectedhomeip/issues/25083
-        // For now, preserve pre-existing behavior, which set credentialType to 1.
-        mCredentials[userIndex][i].credentialType  = CredentialTypeEnum::kPin;
-        mCredentials[userIndex][i].credentialIndex = i + 1;
+        mCredentials[userIndex][i].credentialType  = credentials[i].credentialType;
+        mCredentials[userIndex][i].credentialIndex = credentials[i].credentialIndex;
```

Comment:
Why are these two lines needed?

Reply:
Used memcpy instead.

Reply:
memcpy only reset the first totalCredentials credentials, do we need to clear mCredentials before the memcpy?

Reply:
This part is not an issue, because the the actual thing we store is a Span with the provided length, so we don't look outside of that length.